File: server/api/v1/user/frontend_user_login_log.go
package user

import (
	"github.com/flipped-aurora/gin-vue-admin/server/global"
	"github.com/flipped-aurora/gin-vue-admin/server/model/common/response"
	userReq "github.com/flipped-aurora/gin-vue-admin/server/model/user/request"
	"github.com/gin-gonic/gin"
	"go.uber.org/zap"
)

type FrontendUserLoginLogApi struct{}

// GetFrontendUserLoginLogList 分页获取frontendUserLoginLog表列表
// @Tags FrontendUserLoginLog
// @Summary 分页获取frontendUserLoginLog表列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query userReq.FrontendUserLoginLogSearch true "分页获取frontendUserLoginLog表列表"
// @Success 200 {object} response.Response{data=response.PageResult,msg=string} "获取成功"
// @Router /frontendUserLoginLog/getFrontendUserLoginLogList [get]
func (frontendUserLoginLogApi *FrontendUserLoginLogApi) GetFrontendUserLoginLogList(c *gin.Context) {
	var pageInfo userReq.FrontendUserLoginLogSearch
	err := c.ShouldBindQuery(&pageInfo)
	if err != nil {
		response.FailWithMessage(err.Error(), c)
		return
	}
	list, total, err := frontendUserLoginLogService.GetFrontendUserLoginLogInfoList(pageInfo)
	if err != nil {
		global.GVA_LOG.Error("获取失败!", zap.Error(err))
		response.FailWithMessage("获取失败:"+err.Error(), c)
		return
	}
	response.OkWithDetailed(response.PageResult{
		List:     list,
		Total:    total,
		Page:     pageInfo.Page,
		PageSize: pageInfo.PageSize,
	}, "获取成功", c)
}

File: web/src/api/user/frontendUserLoginLog.js
import service from '@/utils/request'
// @Tags FrontendUserLoginLog
// @Summary 创建frontendUserLoginLog表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.FrontendUserLoginLog true "创建frontendUserLoginLog表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"创建成功"}"
// @Router /frontendUserLoginLog/createFrontendUserLoginLog [post]
export const createFrontendUserLoginLog = (data) => {
  return service({
    url: '/frontendUserLoginLog/createFrontendUserLoginLog',
    method: 'post',
    data
  })
}

// @Tags FrontendUserLoginLog
// @Summary 删除frontendUserLoginLog表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.FrontendUserLoginLog true "删除frontendUserLoginLog表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /frontendUserLoginLog/deleteFrontendUserLoginLog [delete]
export const deleteFrontendUserLoginLog = (params) => {
  return service({
    url: '/frontendUserLoginLog/deleteFrontendUserLoginLog',
    method: 'delete',
    params
  })
}

// @Tags FrontendUserLoginLog
// @Summary 批量删除frontendUserLoginLog表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除frontendUserLoginLog表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /frontendUserLoginLog/deleteFrontendUserLoginLog [delete]
export const deleteFrontendUserLoginLogByIds = (params) => {
  return service({
    url: '/frontendUserLoginLog/deleteFrontendUserLoginLogByIds',
    method: 'delete',
    params
  })
}

// @Tags FrontendUserLoginLog
// @Summary 更新frontendUserLoginLog表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.FrontendUserLoginLog true "更新frontendUserLoginLog表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /frontendUserLoginLog/updateFrontendUserLoginLog [put]
export const updateFrontendUserLoginLog = (data) => {
  return service({
    url: '/frontendUserLoginLog/updateFrontendUserLoginLog',
    method: 'put',
    data
  })
}

// @Tags FrontendUserLoginLog
// @Summary 用id查询frontendUserLoginLog表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query model.FrontendUserLoginLog true "用id查询frontendUserLoginLog表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /frontendUserLoginLog/findFrontendUserLoginLog [get]
export const findFrontendUserLoginLog = (params) => {
  return service({
    url: '/frontendUserLoginLog/findFrontendUserLoginLog',
    method: 'get',
    params
  })
}

// @Tags FrontendUserLoginLog
// @Summary 分页获取frontendUserLoginLog表列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data query request.PageInfo true "分页获取frontendUserLoginLog表列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /frontendUserLoginLog/getFrontendUserLoginLogList [get]
export const getFrontendUserLoginLogList = (params) => {
  return service({
    url: '/frontendUserLoginLog/getFrontendUserLoginLogList',
    method: 'get',
    params
  })
}

// @Tags FrontendUserLoginLog
// @Summary 不需要鉴权的frontendUserLoginLog表接口
// @accept application/json
// @Produce application/json
// @Param data query userReq.FrontendUserLoginLogSearch true "分页获取frontendUserLoginLog表列表"
// @Success 200 {object} response.Response{data=object,msg=string} "获取成功"
// @Router /frontendUserLoginLog/getFrontendUserLoginLogPublic [get]
export const getFrontendUserLoginLogPublic = () => {
  return service({
    url: '/frontendUserLoginLog/getFrontendUserLoginLogPublic',
    method: 'get',
  })
}
